infrastructure/azure.tf

```diff
@@ -102,6 +107,23 @@ resource "azurerm_storage_account" "azurerm_storage_account" {
     }
 }
 
+resource "azurerm_storage_account" "azurerm_functionapp_storage_account" {
+    name                = "${var.azurerm_functionapp_storage_account}"
+    resource_group_name = "${azurerm_resource_group.azurerm_resource_group.name}"
+    location            = "${azurerm_resource_group.azurerm_resource_group.location}"
+
+    # can be one between Premium_LRS, Standard_GRS, Standard_LRS, Standard_RAGRS, Standard_ZRS
+    # see https://docs.microsoft.com/en-us/azure/storage/common/storage-redundancy
+    account_type = "Standard_GRS"
+
+    # see https://docs.microsoft.com/en-us/azure/storage/common/storage-service-encryption
+    enable_blob_encryption = true
+
+    tags {
+        environment = "production"
```

Comment:
I would not set this tag right now as we will probably move the resource creation to a TF module and instantiate the module for the environments (e.g. staging and production), `environment` will be passed as a parameter